### PR TITLE
[GCI] libatomic_ops: Add recipe

### PR DESCRIPTION
**Pull request by CodeforEvolution**
A libatomic_ops recipe made for the google code in. Needs to be fixed though. Google Code-in Task URL: https://codein.withgoogle.com/dashboard/task-instances/6400516531683328/?sp-page=1

### CODE REVIEW

**Comment by CodeforEvolution** [comment]
Ok will check that.

**Comment by KapiX** [comment]
Make whitespace consistent as well.

**Comment by CodeforEvolution** [comment]
Well unfortunately that didn't fix the build.

File: sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+       cd src
+       make install
+       cd ../doc
+       make install
+       cd ../pkgconfig
+       mkdir -p $developDir/lib/pkgconfig/
+       cp atomic_ops.pc $developDir/lib/pkgconfig/
+       cp atomic_ops-uninstalled.pc $developDir/lib/pkgconfig/
+       cd ../
+
```
Comment (KapiX) [inline sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe]: Trailing whitespace

File: sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe [hunk 2]
```diff
@@ -0,0 +1,81 @@
+SUMMARY=“An atomic memory update operations portable implementation"
+DESCRIPTION="Semi-portable access to hardware-provided atomic memory update operations on a number architectures as said on the github project home."
```
Comment (KapiX) [inline sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe]: Break this into lines.

File: sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+HOMEPAGE="https://github.com/ivmai/libatomic_ops/"
+LICENSE="MIT"
+COPYRIGHT="1989, 1991 Free Software Foundation, Inc."
+SOURCE_URI="http://www.ivmaisoft.com/_bin/atomic_ops/libatomic_ops-7.4.2.tar.gz"
+CHECKSUM_SHA256="
```
Comment (KapiX) [inline sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe]: ?

File: sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe [hunk 4]
```diff
@@ -0,0 +1,81 @@
+SUMMARY=“An atomic memory update operations portable implementation"
```
Comment (KapiX) [inline sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe]: Wrong quote character (`“` instead of `"`).

**Comment by KapiX** [comment]
Reorder the blocks according to this: https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#ordering

**Comment by CodeforEvolution** [comment]
Has finally build successfully!

File: sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+       runConfigure ./configure --enable-shared \
+                                --with-gnu-ld \
+                                —-disable-dependency-tracking
+
+       # Can't run make normally since tests won't build.
+       cd src
+       make $jobArgs
+       cd ../doc
+       make $jobArgs
+       cd ../
```
Comment (diversys) [inline sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe]: You don't need this

File: sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+{
+       cd src
+       make install
+       cd ../doc
+       make install
+       cd ../pkgconfig
+       mkdir -p $developDir/lib/pkgconfig/
+       cp atomic_ops.pc $developDir/lib/pkgconfig/
+       cp atomic_ops-uninstalled.pc $developDir/lib/pkgconfig/
+       cd ../
```
Comment (diversys) [inline sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe]: Also not needed

File: sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+       fixPkgconfig
+
+       prepareInstalledDevelLibs \
+              libatomic_ops libatomic_ops_gpl
+
+       # devel package
+       packageEntries devel \
+              $developDir $dataDir
+}
```
Comment (diversys) [inline sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe]: Add new line

File: sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe [hunk 8]
```diff
@@ -0,0 +1,82 @@
+SUMMARY="An atomic memory update operations portable implementation"
+DESCRIPTION="Semi-portable access to hardware-provided \
+             atomic memory update operations on a number architectures \
+             as said on the github project home."
+HOMEPAGE="https://github.com/ivmai/libatomic_ops/"
+COPYRIGHT="1989, 1991 Free Software Foundation, Inc."
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="http://www.ivmaisoft.com/_bin/atomic_ops/libatomic_ops-7.4.2.tar.gz"
+CHECKSUM_SHA256="
```
Comment (KapiX) [inline sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe]: Make it one line.

File: sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+       cmd:make
+       cmd:sed
+       cmd:libtool
+       "
+
+BUILD()
+{
+       runConfigure ./configure --enable-shared \
+                                --with-gnu-ld \
+                                —-disable-dependency-tracking
```
Comment (KapiX) [inline sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe]: `—` is unrecognized by configure.

File: sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+
+       # Can't run make normally since tests won't build.
```
Comment (KapiX) [inline sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe]: I have just built _x86 version and everything worked. Mark x86_gcc2 as broken (put `!` in front of it).

File: sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe [hunk 11]
```diff
@@ -0,0 +1,79 @@
+SUMMARY="An atomic memory update operations portable implementation"
+DESCRIPTION="Semi-portable access to hardware-provided \
+             atomic memory update operations on a number architectures \
```
Comment (KapiX) [inline sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe]: Do not indent descriptions. These spaces will be visible in HaikuDepot.

File: sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+REVISION="1"
+SOURCE_URI="http://www.ivmaisoft.com/_bin/atomic_ops/libatomic_ops-7.4.2.tar.gz"
+CHECKSUM_SHA256="04fa615f62992547bcbda562260e28b504bc4c06e2f985f267f3ade30304b5dd"
+
+ARCHITECTURES="x86_gcc2 x86"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+       libatomic_ops$secondaryArchSuffix = $portVersion
+       lib:libatomic_ops$secondaryArchSuffix = $portVersion
```
Comment (KapiX) [inline sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe]: Change `$portVersion` to `1.0.3 compat >= 1`.

File: sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+       lib:libatomic_ops_gpl$secondaryArchSuffix = $portVersion
```
Comment (KapiX) [inline sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe]: Ditto.

File: sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+       "
+REQUIRES="
+       haiku$secondaryArchSuffix
+       "
+
+PROVIDES_devel="
+       libatomic_ops$secondaryArchSuffix_devel = $portVersion
+       devel:libatomic_ops$secondaryArchSuffix = $portVersion
```
Comment (KapiX) [inline sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe]: Ditto.

File: sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+       devel:libatomic_ops_gpl$secondaryArchSuffix = $portVersion
```
Comment (KapiX) [inline sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe]: Ditto.

File: sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe [hunk 16]
```diff
@@ -0,0 +1,79 @@
+SUMMARY="An atomic memory update operations portable implementation"
+DESCRIPTION="Semi-portable access to hardware-provided \
+             atomic memory update operations on a number architectures \
+             as said on the github project home."
+HOMEPAGE="https://github.com/ivmai/libatomic_ops/"
+COPYRIGHT="1989, 1991 Free Software Foundation, Inc."
```
Comment (KapiX) [inline sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe]: You have put copyright of the license here, not of the code.
By inspecting the source files, I have found that the proper copyright should be `Hewlett-Packard Development Company, L.P.`. Try to find the year range.

File: sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+LICENSE="MIT"
```
Comment (KapiX) [inline sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe]: It is not MIT licensed code. It's GPLv2.

**Comment by CodeforEvolution** [comment]
Would the recipe now be considered ready?

File: sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe [hunk 18]
```diff
@@ -0,0 +1,70 @@
...
+       lib:libatomic_ops$secondaryArchSuffix = 1.0.3 compat >= 1
+       lib:libatomic_ops_gpl$secondaryArchSuffix = 1.0.3 compat >= 1
+       "
+REQUIRES="
+       haiku$secondaryArchSuffix
+       "
+
+PROVIDES_devel="
+       libatomic_ops$secondaryArchSuffix_devel = 1.0.3 compat >= 1
+       libatomic_ops_x86$secondaryArchSuffix_devel = 1.0.3 compat >= 1
```
Comment (KapiX) [inline sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe]: This expands to `libatomic_ops_x86_x86`. Remove it.

File: sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe [hunk 19]
```diff
@@ -0,0 +1,70 @@
...
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="http://www.ivmaisoft.com/_bin/atomic_ops/libatomic_ops-7.4.2.tar.gz"
+CHECKSUM_SHA256="04fa615f62992547bcbda562260e28b504bc4c06e2f985f267f3ade30304b5dd"
+
+ARCHITECTURES="!x86_gcc2 x86"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+       libatomic_ops$secondaryArchSuffix = 1.0.3 compat >= 1
```
Comment (KapiX) [inline sys-libs/libatomic_ops/libatomic_ops-7.4.2.recipe]: You were supposed to change `$portVersion` only for provided libraries. This entry should still have `$portVersion`.

**Comment by CodeforEvolution** [comment]
Ready now?

**Comment by KapiX** [comment]
Yes, now please squash your commits.
https://github.com/haikuports/haikuports/wiki/DevelopmentModel#how-do-i-make-one-final-commit-from-many-for-the-pull-request-pr

**Comment by CodeforEvolution** [comment]
I'm having some trouble with this part. May need some help.

**Comment by KapiX** [comment]
1. Copy your recipe somewhere safe.
2. `git reset --hard f732d827658cf433883feb24e4af56f2b0243f10` This hash points to the commit before your commits.
3. Copy your recipe into the tree again.
4. Add, commit and `push -f`.

**Comment by CodeforEvolution** [comment]
Ok, I have followed your steps, anything else I need to do?

**Comment by KapiX** [comment]
Yes, submit your work in the task :)

**Comment by CodeforEvolution** [comment]
Unfortunately the contest closed so I can't submit it. Though thank you for all your help and in the future thanks to you, it most likely won't take 12 commits to make a recipe for haiku :).

**Comment by KapiX** [comment]
That's a shame. Thanks for your work :)

**Pull request merged**